packages/library/src/rpc-integer-overflow-error.ts

```diff
@@ -1,45 +1,45 @@
-            const lastTwoDigits = argPosition % 100;
-            if (lastDigit == 1 && lastTwoDigits != 11) {
-                argumentLabel = argPosition + 'st';
-            } else if (lastDigit == 2 && lastTwoDigits != 12) {
-                argumentLabel = argPosition + 'nd';
-            } else if (lastDigit == 3 && lastTwoDigits != 13) {
-                argumentLabel = argPosition + 'rd';
-            } else {
-                argumentLabel = argPosition + 'th';
-            }
+export function createSolanaJsonRpcIntegerOverflowError(
+    methodName: string,
+    keyPath: KeyPath,
+    value: bigint,
+): SolanaError<typeof SOLANA_ERROR__RPC_INTEGER_OVERFLOW> {
+    let argumentLabel = '';
+    if (typeof keyPath[0] === 'number') {
+        const argPosition = keyPath[0] + 1;
+        const lastDigit = argPosition % 10;
+        const lastTwoDigits = argPosition % 100;
+        if (lastDigit == 1 && lastTwoDigits != 11) {
+            argumentLabel = argPosition + 'st';
+        } else if (lastDigit == 2 && lastTwoDigits != 12) {
+            argumentLabel = argPosition + 'nd';
+        } else if (lastDigit == 3 && lastTwoDigits != 13) {
+            argumentLabel = argPosition + 'rd';
         } else {
-            argumentLabel = `\`${keyPath[0].toString()}\``;
+            argumentLabel = argPosition + 'th';
         }
-        const path =
-            keyPath.length > 1
-                ? keyPath
-                      .slice(1)
-                      .map(pathPart => (typeof pathPart === 'number' ? `[${pathPart}]` : pathPart))
-                      .join('.')
-                : null;
-        super(
-            `The ${argumentLabel} argument to the \`${methodName}\` RPC method` +
-                `${path ? ` at path \`${path}\`` : ''} was \`${value}\`. This number is ` +
-                'unsafe for use with the Solana JSON-RPC because it exceeds ' +
-                '`Number.MAX_SAFE_INTEGER`.',
-        );
-        this.keyPath = keyPath;
-        this.methodName = methodName;
-        this.value = value;
+    } else {
+        argumentLabel = `\`${keyPath[0].toString()}\``;
     }
-    get name() {
-        return 'SolanaJsonRpcIntegerOverflowError';
+    const path =
+        keyPath.length > 1
+            ? keyPath
+                  .slice(1)
+                  .map(pathPart => (typeof pathPart === 'number' ? `[${pathPart}]` : pathPart))
+                  .join('.')
+            : undefined;
+    const error = new SolanaError(SOLANA_ERROR__RPC_INTEGER_OVERFLOW, {
+        argumentLabel,
+        keyPath: keyPath as readonly (string | number | symbol)[],
+        methodName,
+        optionalPathLabel: path ? ` at path \`${path}\`` : '',
+        value,
+        ...(path !== undefined ? { path } : undefined),
+    });
+    if ('captureStackTrace' in Error && typeof Error.captureStackTrace === 'function') {
+        Error.captureStackTrace(error, createSolanaJsonRpcIntegerOverflowError);
```

Comment:
This might be interesting to you, viewer. Lets you truncate the stack trace at {some method} so that it doesn't _appear_ in the stack trace. It's not interesting that the error happened to be `newed` up in `createSolanaJsonRpcIntegerOverflow()` – all the user cares about is where the error was thrown.